contracts/account/utils/draft-ERC7579Utils.sol

```diff
@@ -169,12 +172,40 @@ library ERC7579Utils {
     }
 
     /// @dev Decodes a batch of executions. See {encodeBatch}.
+    ///
+    /// NOTE: This function runs some checks and will throw a {ERC7579DecodingError} if the input is not properly formatted.
     function decodeBatch(bytes calldata executionCalldata) internal pure returns (Execution[] calldata executionBatch) {
-        assembly ("memory-safe") {
-            let ptr := add(executionCalldata.offset, calldataload(executionCalldata.offset))
-            // Extract the ERC7579 Executions
-            executionBatch.offset := add(ptr, 32)
-            executionBatch.length := calldataload(ptr)
+        unchecked {
+            uint256 bufferLength = executionCalldata.length;
+
+            // Check executionCalldata is not empty.
+            if (bufferLength < 32) revert ERC7579DecodingError();
+
+            // Get the offset of the array, and check its valid.
+            uint256 offset = uint256(bytes32(executionCalldata[0:32]));
+
+            // Array length is between offset and offset + 32. We check that this is in the buffer.
+            // Since we know executionCalldata is at least 32, we can subtract with no overflow risk.
+            // This also guarantees that offset + 32 doesn't overflow.
+            if (offset > bufferLength - 32) revert ERC7579DecodingError();
+
+            // Get the array length.
+            uint256 arrayLength = uint256(bytes32(executionCalldata[offset:offset + 32]));
+            if (arrayLength > type(uint64).max) revert ERC7579DecodingError();
+
+            // Get the array as a bytes slice, and check its is long enough:
+            // - `96` is the size of `Execution`'s "head" representation. This is composed of
+            //   - 32 bytes for the `target` (address)
+            //   - 32 bytes for the `value` (uint256)
+            //   - 32 bytes for the offset pointer to the `callData` (bytes)
+            // - `arrayLength * 96` does not overflow because `arrayLength` is less than `2**64`.
+            bytes calldata executionArray = executionCalldata[offset + 32:];
+            if (executionArray.length < arrayLength * 96) revert ERC7579DecodingError();
```

Comment:
See https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5353/files#r1881380537

Which I was able to confirm 
```
ethers.AbiCoder.defaultAbiCoder().encode(["(address,uint256,bytes)[]"], [[["0xA0c32d8A1B956Cfa83AeE3E6D2afBd0b9c2a188F", 1, "0x01234567"], ["0xA0c32d8A1B956Cfa83AeE3E6D2afBd0b9c2a188F", 2, "0x89abcdef"]]])


0x
0x0000 0000000000000000000000000000000000000000000000000000000000000020 // offset to the start of the array (0x0000 + 0x20 => 0x0020)
0x0020 0000000000000000000000000000000000000000000000000000000000000002 // length of the array (2)
[begining of the array]
0x0040 0000000000000000000000000000000000000000000000000000000000000040 // offset to element #0 (0x0040 + 0x40 = 0x0080) 
0x0060 00000000000000000000000000000000000000000000000000000000000000e0 // offset to element #1 (0x0040 + 0xe0 = 0x0120)
[element #0]
0x0080 000000000000000000000000a0c32d8a1b956cfa83aee3e6d2afbd0b9c2a188f // target
0x00a0 0000000000000000000000000000000000000000000000000000000000000001 // value
0x00c0 0000000000000000000000000000000000000000000000000000000000000060 // offset to calldata (0x0080 + 0x60 = 0x00e0)
[data for element #0]
0x00e0 0000000000000000000000000000000000000000000000000000000000000004 // length
0x0100 0123456700000000000000000000000000000000000000000000000000000000 // content
[element #1]
0x0120 000000000000000000000000a0c32d8a1b956cfa83aee3e6d2afbd0b9c2a188f // target
0x0140 0000000000000000000000000000000000000000000000000000000000000002 // value
0x0160 0000000000000000000000000000000000000000000000000000000000000060 // offset to calldata (0x0120 + 0x60 = 0x0180)
[data for element #1]
0x0180 0000000000000000000000000000000000000000000000000000000000000004 // length
0x01a0 89abcdef00000000000000000000000000000000000000000000000000000000 // content
```

```suggestion
            // Get the array as a bytes slice, and check its is long enough:
            // - each element of the array is an "offset pointer" to the data
            //   - each offset pointer takes 32 bytes
            //   - validity of the calldata at that location is checked when the array element is accessed.
            // - `arrayLength * 32` does not overflow because `arrayLength` is less than `2**64`.
            bytes calldata executionArray = executionCalldata[offset + 32:];
            if (executionArray.length < arrayLength * 32) revert ERC7579DecodingError();
```

This breaks some of the tests written by @frangio :/

Reply:
Fixed in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5353/commits/6aaa56acdebea0c2b113552b3a287619f0f94406